source/OptimaJet.DWKit.StarterApplication/Services/GroupService.cs

```diff
@@ -0,0 +1,80 @@
+using JsonApiDotNetCore.Data;
+using JsonApiDotNetCore.Internal;
+using JsonApiDotNetCore.Services;
+using Microsoft.Extensions.Logging;
+using Optimajet.DWKit.StarterApplication.Models;
+using OptimaJet.DWKit.StarterApplication.Repositories;
+
+namespace OptimaJet.DWKit.StarterApplication.Services
+{
+    public class GroupService : EntityResourceService<Group>
+    {
+        public IOrganizationContext OrganizationContext { get; private set; }
+        public IJsonApiContext JsonApiContext { get; }
+        public ICurrentUserContext CurrentUserContext { get; }
+        public UserRepository UserRepository { get; }
+        public GroupRepository GroupRepository { get; }
+
+        public GroupService(
+            IJsonApiContext jsonApiContext,
+            IOrganizationContext organizationContext,
+            ICurrentUserContext currentUserContext,
+            UserRepository userRepository,
+            IEntityRepository<Group> groupRepository,
+            ILoggerFactory loggerFactory) : base(jsonApiContext, groupRepository, loggerFactory)
+        {
+            OrganizationContext = organizationContext;
+            JsonApiContext = jsonApiContext;
+            CurrentUserContext = currentUserContext;
+            UserRepository = userRepository;
+            GroupRepository = (GroupRepository)groupRepository;
+        }
+
+
+        public override async Task<IEnumerable<Group>> GetAsync()
+        {
+            if (OrganizationContext.SpecifiedOrganizationDoesNotExist)
+            {
+                return Enumerable.Empty<Group>().AsQueryable();
+            }
+            else
+            {
+                var query = JsonApiContext.QuerySet;
+                if (query == null)
+                {
+                    query = new JsonApiDotNetCore.Internal.Query.QuerySet();
+                    JsonApiContext.QuerySet = query;
+                }
+                var value = OrganizationContext.HasOrganization ? OrganizationContext.OrganizationId.ToString() : "";
+                query.Filters.Add(new JsonApiDotNetCore.Internal.Query.FilterQuery("organization-header", value, "="));
+                return await base.GetAsync();
+            }
+        }
+        public override async Task<Group> GetAsync(int id)
+        {
+            var groups = await GetAsync();
+            return groups.SingleOrDefault(g => g.Id == id);
+        }
+        public override async Task<Group> UpdateAsync(int id, Group resource)
+        {
+            var updateForm = new Forms.Groups.UpdateForm(UserRepository,
+                                                         GroupRepository, 
+                                                         CurrentUserContext);
+            if (!updateForm.IsValid(id, resource))
+            {
+                throw new JsonApiException(updateForm.Errors);
+            }
+            // Make sure that group may be accessed by current user
+            var group = await GetAsync(id);
```

Comment:
what does this comment have to do with the code immediately below it? The current user is not involved here.

Reply:
This is another case where a get was being performed so that the filters would run and make sure that the current user was a member or the group owner id and that the group owner matched the organization header  entry if it was set to a value.  For updates, that has now been moved to the base validation form.

Reply:
ok cool. 

It's just a comment should be directly making things _more_ clear, or the comment isn't needed / worth having :)